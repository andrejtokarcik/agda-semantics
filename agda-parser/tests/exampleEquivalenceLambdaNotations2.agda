module exampleEquivalenceLambdaNotations2 where


postulate A : Set

g : A -> A
g = \x -> x

g' : A -> A
g' a = a

postulate P : (A -> A) -> Set

f : P g -> P g'
f x = x
