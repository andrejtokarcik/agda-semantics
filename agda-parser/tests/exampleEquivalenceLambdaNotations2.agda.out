<k>
    .K
</k>
<mgu>
    subst(.KList)
</mgu>
<ctxstack>
    .List
</ctxstack>
<sig>
    Fun ( "f" ) |-> ( #symVariable(21) : (Name ( "P" ) Fun ( "g" )) ) -> (Name
       ( "P" ) Fun ( "g'" ))
    Fun ( "g" ) |-> ( Var ( "x" ) : Name ( "A" ) ) -> Name ( "A" )
    Fun ( "g'" ) |-> ( Var ( "x" ) : Postulate ( "A" ) ) -> Postulate ( "A" )
    'Postulate("A") |-> Set ( 0 )
    'Postulate("P") |-> ( #symVariable(17) : ( Var ( "x" ) : Postulate ( "A" )
       ) -> Postulate ( "A" ) ) -> Set ( 0 )
</sig>
<con>
    .Map
</con>
<ctx>
    .Map
</ctx>
<meta>
    .Map
</meta>
<ncat>
    Name ( "A" ) |-> 'Postulate("A")
    Name ( "P" ) |-> 'Postulate("P")
    Name ( "f" ) |-> Fun ( "f" )
    Name ( "g" ) |-> Fun ( "g" )
    Name ( "g'" ) |-> Fun ( "g'" )
</ncat>
<fun>
    Fun ( "f" ) |-> \ ( Var ( "x" ) : #symExpr(24) ) -> Var ( "x" )
    Fun ( "g" ) |-> \ ( Var ( "x" ) : Postulate ( "A" ) ) -> Var ( "x" )
    Fun ( "g'" ) |-> \ ( Var ( "x" ) : Postulate ( "A" ) ) -> Var ( "x" )
</fun>
